src/Stratis.SmartContracts.Executor.Reflection/Contract.cs

```diff
@@ -103,13 +130,34 @@ public IContractInvocationResult Invoke(string methodName, IReadOnlyList<object>
             if (methodToInvoke.IsPrivate)
                 return ContractInvocationResult.Failure(ContractInvocationErrorType.MethodIsPrivate);
 
-            // Restore the state of the instance
-            if(!this.initialized)
-                SetStateFields(this.instance, this.State);
+            EnsureInitialized();
 
             return this.InvokeInternal(methodToInvoke, invokeParams);
         }
 
+        private IContractInvocationResult InvokeFallback()
+        {
+            // Handles the scenario where no fallback was defined, but it is attempted to be invoked anyway.
+            // This could occur if a method invocation is directly made to the fallback via a transaction.
+            if (this.Fallback == null)
+                return ContractInvocationResult.Failure(ContractInvocationErrorType.MethodDoesNotExist);
+
+            EnsureInitialized();
+
+            return this.InvokeInternal(this.Fallback, null);
```

Comment:
Could be worth looking into just doing instance.Fallback() if we stick with that approach.